.devcontainer/Dockerfile

```diff
@@ -120,18 +120,18 @@ RUN echo "Install 3rd party dependencies" && \
         grpc-dev
 
 ##### Useful for logfile modification e.g. pruning all /magma/... prefix from GCC warning logs
-RUN GOBIN="/usr/bin/" go install github.com/ezekg/xo@0f7f076932dd
+RUN GOBIN="/usr/bin/" go install github.com/ezekg/xo@0f7f076932dd && \
+    rm --recursive --interactive=never /root/.cache/go-build
 
 ##### GRPC and it's dependencies
 RUN git clone --recurse-submodules -b v1.35.0 https://github.com/grpc/grpc && \
-    cd grpc && \
-    mkdir -p cmake/build && \
-    cd cmake/build && \
+    mkdir -p grpc/cmake/build && \
+    cd grpc/cmake/build && \
     cmake -DgRPC_INSTALL=ON -DgRPC_BUILD_TESTS=OFF -DBUILD_SHARED_LIBS=ON ../.. && \
     make -j"$(nproc)" && \
     make install && \
-    cd / && \
-    rm -rf grpc
+    cd ../../.. && \
```

Comment:
This solution has the disadvantages that we have to count how many directories we descended. I discussed with @nstng to use `pushd`/`popd`, or running the `cd` in a subshell so that the parent shell never changes directories. We figured that this version is the most widely understood one and thus kept it despite the disadvantage of having to count.

Reply:
How about introducing a variable for the workdir? I.e.
```dockerfile
ARG basedir=/usr/local
cd where/ever/you/want
...
cd ${basedir}
```
This would avoid the counting and be pretty safe to always return to the desired directory.
Similar to https://docs.docker.com/engine/reference/builder/#workdir.

Reply:
I would prefer the current solution as it is a bit less stateful as it doesn't care what the working directory is set to.

Reply:
Another thought: How about the "poor mans pushd/popd": `cd -`? Might that be a valid option?

Reply:
That wouldn't work for `prometheus-cpp` below, as that uses multiple `cd` commands. I'm inclined to not put more thought into this, but accept it as good enough.